commit d3be3e0e58e62e9e941546b28c294084e622c2ae (refs/changes/39/3239/1)
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2018-01-19T10:22:31-07:00 (1 year, 9 months ago)
    
    MANTA-3564 Muppet no_backends amon probe does not work correctly

diff --git a/alarm_metadata/probe_templates/loadbalancer.yaml b/alarm_metadata/probe_templates/loadbalancer.yaml
index b1548b1..aa90de8 100644
--- a/alarm_metadata/probe_templates/loadbalancer.yaml
+++ b/alarm_metadata/probe_templates/loadbalancer.yaml
@@ -133,10 +133,11 @@
         -
             type: cmd
             config:
-                cmd: "test -n $(/opt/smartdc/muppet/build/node/bin/node /opt/smartdc/muppet/node_modules/.bin/haproxystat showStat /tmp/haproxy  | /opt/smartdc/muppet/build/node/bin/node -e 's=\"\"; process.stdin.resume(); process.stdin.on(\"data\",function(c){s+=c}); process.stdin.on(\"end\",function(){o=eval(\"(\"+s+\")\");console.log(JSON.stringify(o)); });' | /usr/bin/json  -c 'this.type == \"backend\" && !/stats/.test(this.pxname) && act == 0' -a pxname)"
-                interval: 30
+                cmd: "test $(/opt/smartdc/muppet/build/node/bin/node /opt/smartdc/muppet/node_modules/.bin/haproxystat showStat /tmp/haproxy  | /opt/smartdc/muppet/build/node/bin/node -e 's=\"\"; process.stdin.resume(); process.stdin.on(\"data\",function(c){s+=c}); process.stdin.on(\"end\",function(){o=eval(\"(\"+s+\")\");console.log(JSON.stringify(o)); });' | /usr/bin/json  -c 'this.type == \"backend\" && !/stats/.test(this.pxname) && act == 0' -a pxname | wc -l) -eq 0"
+                timeout: 10
+                interval: 60
                 threshold: 2
-                period: 120
+                period: 300
     ka:
         title: Loadbalancer has no backends
         description: A loadbalancer has no working backends
